### PR TITLE
Update foreman, foreman-proxy & katello dependencies

### PR DESCRIPTION
**Pull request by ekohl**
foreman and foreman-proxy use rich dependencies, but katello is gem based and gem2rpm doesn't understand that yet.

### CODE REVIEW

File: packages/foreman/foreman/foreman.spec [hunk 1]
```diff
@@ -49,83 +49,50 @@ Requires: (%{name}-fapolicyd if fapolicyd)
...
-Requires: rubygem(rack-cors) < 2.0
-Requires: rubygem(jwt) >= 2.2.2
-Requires: rubygem(jwt) < 3.0
-Requires: rubygem(graphql) >= 1.13.0
-Requires: rubygem(graphql) < 1.14.0
+Requires: (rubygem(rack-cors) >= 1.1 with rubygem(rack-cors) < 2.0)
+Requires: (rubygem(jwt) >= 2.2.2 with rubygem(jwt) < 3.0)
+Requires: (rubygem(graphql) >= 1.13.0 with rubygem(graphql) < 1.14.0)
 Requires: rubygem(graphql-batch)
+Requires: rubygem(rss)
```
Comment (evgeni) [inline packages/foreman/foreman/foreman.spec:95]: nope? :)

File: packages/foreman/foreman/foreman.spec [hunk 2]
```diff
@@ -149,83 +116,50 @@ BuildRequires: rubygem(rake) >= 0.8.3
...
-BuildRequires: rubygem(rack-cors) < 2.0
-BuildRequires: rubygem(jwt) >= 2.2.2
-BuildRequires: rubygem(jwt) < 3.0
-BuildRequires: rubygem(graphql) >= 1.13.0
-BuildRequires: rubygem(graphql) < 1.14.0
+BuildRequires: (rubygem(rack-cors) >= 1.1 with rubygem(rack-cors) < 2.0)
+BuildRequires: (rubygem(jwt) >= 2.2.2 with rubygem(jwt) < 3.0)
+BuildRequires: (rubygem(graphql) >= 1.13.0 with rubygem(graphql) < 1.14.0)
 BuildRequires: rubygem(graphql-batch)
+BuildRequires: rubygem(rss)
```
Comment (evgeni) [inline packages/foreman/foreman/foreman.spec:162]: nope

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 3]
```diff
@@ -49,18 +47,18 @@ BuildRequires: rubygem(apipie-rails) >= 0.5.14
...
-BuildRequires: rubygem(pulpcore_client) >= 3.28.0
-BuildRequires: rubygem(pulpcore_client) < 3.29.0
-BuildRequires: rubygem(pulp_file_client) >= 1.14.0
-BuildRequires: rubygem(pulp_file_client) < 1.15.0
-BuildRequires: rubygem(pulp_ansible_client) >= 0.18.0
-BuildRequires: rubygem(pulp_ansible_client) < 0.19.0
-BuildRequires: rubygem(pulp_container_client) >= 2.15.0
-BuildRequires: rubygem(pulp_container_client) < 2.16.0
+BuildRequires: rubygem(pulpcore_client) >= 3.39.0
+BuildRequires: rubygem(pulpcore_client) < 3.40.0
```
Comment (evgeni) [inline packages/katello/rubygem-katello/rubygem-katello.spec:51]: no `with` here?

Comment (evgeni) [reply packages/katello/rubygem-katello/rubygem-katello.spec:51]: ah, this is because these come from gem2rpm and that didn't work you said, kk

Comment (ekohl) [reply packages/katello/rubygem-katello/rubygem-katello.spec:51]: exactly - I'm trying to figure out how to do this as cleanly as possible (and later submit an update to gem2rpm).

File: packages/foreman/foreman/foreman.spec (hunk 1 above)
Comment (ekohl) [reply packages/foreman/foreman/foreman.spec:95]: Ah yes, my local gemfile for Ruby 3 strikes again.

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 4]
```diff
@@ -22,17 +22,13 @@ Requires: (katello-selinux if selinux-policy-targeted)
...
-Requires: ruby < 3
 BuildRequires: foreman-assets >= %{foreman_min_version}
 BuildRequires: foreman-plugin >= %{foreman_min_version}
 Requires: ruby >= 2.5
 Requires: ruby < 3
 BuildRequires: ruby >= 2.5
 BuildRequires: ruby < 3
-BuildRequires: rubygems-devel
-BuildArch: noarch
-Provides: foreman-plugin-%{plugin_name} = %{version}
```
Comment (evgeni) [inline packages/katello/rubygem-katello/rubygem-katello.spec:35]: removing those two seems wrong?

Comment (ekohl) [reply packages/katello/rubygem-katello/rubygem-katello.spec:35]: If you look carefully it's moved around, to lines 59/60. I debated moving it in the template to make the diff smaller.

Comment (evgeni) [reply packages/katello/rubygem-katello/rubygem-katello.spec:35]: aha! damn it diff view!

**Review by evgeni — APPROVED** [review]

**Pull request merged**